### PR TITLE
remove inclusion phrases in MainQueryParser

### PR DESCRIPTION
**Pull request by yingjieMiao**

### CODE REVIEW

File: shoebox/app/com/keepit/search/MainQueryParser.scala [hunk 1]
```diff
@@ -70,7 +71,7 @@ class MainQueryParser(
         query.setBoost(baseBoost)
 
         val phrases = if (numStemmedTerms > 1 && (phraseBoost > 0.0f || phraseProximityBoost > 0.0f)) {
-          phraseDetector.detectAll(getStemmedTermArray)
+          RemoveOverlapping.removeInclusions(phraseDetector.detectAll(getStemmedTermArray))
```
Comment (ymatsuda) [inline shoebox/app/com/keepit/search/MainQueryParser.scala:74]: An alternative is to add detectNoOverlapping method to PhraseDetector and do overlap removal internally.

Comment (yingjieMiao) [inline shoebox/app/com/keepit/search/MainQueryParser.scala:74]: True. But that will change phraseQueryTest too :)

I thought having a filter outside is more modular and transparent. 

Later, we may implement a more intelligent filter.

Comment (ymatsuda) [inline shoebox/app/com/keepit/search/MainQueryParser.scala:74]: I'd rather encapsulate such details in the detector.

Comment (yingjieMiao) [inline shoebox/app/com/keepit/search/MainQueryParser.scala:74]: Sure. I can do that.

**Pull request merged**